### ISSUE TITLE
Accessibility: Plugin Single Use of Color Failures

### ISSUE THREAD

**Issue by amberhinds** [WordPress/wordpress.org#262]
On the plugin single page ([Gutenberg plugin test page](https://test.wordpress.org/plugins/gutenberg/)), there are several instances where links are indicated only by a change of color and do not have any other way to indicate that they are links. This is a failure of [WCAG 1.4.1 Use of Color (A)](https://www.w3.org/TR/WCAG22/#use-of-color).

Here are the instances I found:

1. "See all 56" - referring to languages in plugin meta:
<img width="338" alt="Screenshot 2024-04-16 at 5 10 29 PM" src="https://github.com/WordPress/wordpress.org/assets/13153456/b12c8376-224f-49ee-8f81-a9ce00ed916c">

2. The Advanced View link:
<img width="349" alt="Screenshot 2024-04-16 at 5 10 55 PM" src="https://github.com/WordPress/wordpress.org/assets/13153456/aca5d0e1-f5e6-4652-a5a8-2121c5da8cd1">

3. All of the rating links (both the number of stars "5 stars" and the count "821"):
<img width="319" alt="Screenshot 2024-04-16 at 5 12 18 PM" src="https://github.com/WordPress/wordpress.org/assets/13153456/2de6563d-d6ee-4ba6-a91a-1813211a9be0">

4. Contributors & Developer links:
<img width="651" alt="Screenshot 2024-04-16 at 5 13 04 PM" src="https://github.com/WordPress/wordpress.org/assets/13153456/45ec59b2-f0b3-4bdf-b3e3-6ea2bf4c1135">

Links need a second way, aside from color to indicate that they are links and actionable. Traditionally this is done with underlines, but arrows or other icons can also be used as an alternative.

**Comment by ryelle** [comment]
I thought that only applies to links inside content? That these links, like the navigation links, were understood to be links by the context around them (being in the sidebar as "meta" info — the exception might be example 4). They also have underlines on hover, to help clarify.

For example, this pattern exists on [Learn](https://learn.wordpress.org/lesson-plan/de-how-to-find-theme-specific-documentation/) and [Showcase](https://wordpress.org/showcase/tonal/), no underlines in the sidebar content until hover.

<img width="350" alt="Screenshot 2024-04-16 at 6 24 42 PM" src="https://github.com/WordPress/wordpress.org/assets/541093/aa846bca-84cb-4761-af9d-d748737181d9"> <img width="389" alt="Screenshot 2024-04-16 at 6 24 46 PM" src="https://github.com/WordPress/wordpress.org/assets/541093/86de9b7a-71b1-4f9d-b028-392b9c198671">

**Comment by amberhinds** [comment]
It applies to links where you cannot clearly differentiate that it is a link. In the first example, maybe those are okay because they are all links and are a different weight from the text on the left. But in the second example that you posted, if you couldn't differentiate between the color blue and black, how would you know that "Business" is a link but "United States" is not?

**Comment by amberhinds** [comment]
Essentially, people shouldn't have to hover over something to know that it's a link. Sometimes, positioning is enough to make it stand out as a link. But having some elements in a "table" (I know those are not tables, they just look like them) that are linked and others that are not, it's not sufficient to only use color to indicate which are links and which are not.

**Comment by ryelle** [comment]
Okay, then I'll tag in @WordPress/meta-design since it sounds like we might need to revisit link styles across the entire redesign 😬

**Comment by StevenDufresne** [comment]
I'm not pushing back against this but interested in learning more.

For example, from the link you shared
> [WCAG 1.4.1 Use of Color (A)](https://www.w3.org/TR/WCAG22/#use-of-color).

The left-hand menu seems to also violate the rule based on the [explanation](https://github.com/WordPress/wordpress.org/issues/262#issuecomment-2060071211).

<img width="431" alt="Screenshot 2024-04-17 at 9 03 21 AM" src="https://github.com/WordPress/wordpress.org/assets/1657336/854e20d8-017e-4dca-94d3-24304a243810">

Those are all links that are only discoverable after hover.

**Comment by amberhinds** [comment]
The difference there is the context. 

It's under a heading of "Table of Contents" which in and of itself may communicate to people (in a web scenario) that the items below the heading are all clickable. I think that communicates to people an expectation of functionality.

Also, every single thing in that list, styled that way is a link. There's not a mix of links and non-links that share the same styles in that list.

What I was getting at with my first comment is that it might be OK in some scenarios not to underline or have an icon if there's some other visual distinction between the link and adjacent text. While I don't love this, it might get a "pass" from some testers because everything on the right is a link and the links have a lower font-weight than the text at the left.

![image](https://github.com/WordPress/wordpress.org/assets/13153456/7b079fda-b7f0-4f1b-8423-56185d04d7e8)

It's most problematic in scenarios like this where it's not clear if something is a link or not:
![image](https://github.com/WordPress/wordpress.org/assets/13153456/e7a9513a-2fb4-4901-aa08-4fc586e01def)

People might expect to be able to click on the author name or country. Or vice versa, if the first couple are not links and I can't distinguish the blue from the black, I wouldn't expect that I could click on the category or flavor.

**Comment by ndiego** [comment]
Thanks for flagging this issue and the additional context 🙏

While we should definitely investigate and look into a design solution asap, I personally don't think it should block launching the new Plugins design since the link behavior is the same in the current theme (and elsewhere on the the site).

**Comment by fcoveram** [comment]
It could be a solution adding context to the area rather than modifying the font weight or adding an underline style? Given that the WCAG documentation example meets the requirement by clarifying the section purpose through a heading, we could include one in this section noting the meta content.

**Comment by joedolson** [comment]
It's pretty much entirely about whether a link is mixed with non-linked content. Even with a heading, the case with Author/Country/Category/Flavor/Published/Site tags posted above would fail, because some of those strings are linked and others aren't, and there's no way to distinguish between them. 

The use of color criteria is about differentiating items: in this case, is it possible to differentiate a link from a non-link?

In the Table of Contents example from the W3C, there are several indicators that those are links: the heading 'Table of Contents', the background color that sets the content off from neighboring content, etc., 

If some of the text following the Table of Contents heading was linked and some of it wasn't, there would definitely be a problem there. 

Following that logic, the example from Learn can pass; but the example from Showcase can't. It might be better if the Learn example had underlines, but I wouldn't actually call it a WCAG failure; being an accessibility problem and being a WCAG failure are two different things.
